src/core/AnimatedArrayFrom.js

```diff
@@ -0,0 +1,50 @@
+import invariant from 'fbjs/lib/invariant';
+
+import AnimatedNode from './AnimatedNode';
+import { val, adapt } from '../utils';
+
+const mem = {};
+
+export default function memoizeArrayFromNode(arr) {
+  invariant(
+    arguments.length === 1 && Array.isArray(arr),
+    'node arrayFrom expect single array as argument'
+  );
+
+  const hash = arr
+    .map(x => (x instanceof AnimatedNode ? x.__nodeID : 'e'))
+    .join('s');
+  /**
+   * We are adding 'e' for all arbitrary values to prevent cases when the same
```

Comment:
This hash seems to be useful, but I cannot se point of this `e`. IMHO  `x` should be always instance of `AnimatedNode`.

Reply:
I'm using `adapt` after that, not everything is AnimatedNode at this point.
I think that there's no point in comparing nodeID of arbitrary values since they can change.

Reply:
Okay, right